sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3560,6 +3560,18 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-32237: Hint in CTE") {
```

Comment:
Like JoinHintSuite, we should have our own hint-specific suite. It can help us understand the test coverage of SQL HINT.

Reply:
Sure. I will add a new suite.